h5pyd/_apps/utillib.py

```diff
@@ -691,88 +816,8 @@ def update_chunktable(src, tgt, ctx):
-        return
-    # prior to HSDS v0.7.0+, reads failed if str was used for s3path,
-    # store as bytes
-    s3path = ctx["s3path"].encode("utf-8")
-    dt = get_chunktable_dtype(include_file_uri=extend)
-    chunkinfo_arr = np.zeros(chunktable_dims, dtype=dt)
-    rank = len(chunktable_dims)
-    num_chunks = get_num_chunks(src)
-   
-    if is_h5py(src):   
-        if src.chunks is None:
-            chunk_offset = get_dset_offset(src)
-            if chunk_offset <= 0:
-                msg = "Expected dset_offset to be greater than zero"
-                logging.error(msg)
-                if not ctx["ignore_error"]:
-                    raise IOError(msg)
-                return
-
-            chunk_size = src.id.get_storage_size()
-            if extend:
-                chunkinfo_arr[...] = (chunk_offset, chunk_size, s3path)
-            else:
-                chunkinfo_arr[...] = (chunk_offset, chunk_size)
-        else:
-            spaceid = src.id.get_space()
-
-            if library_has_chunk_iter:
-                def update_chunktable_callback(chunk_info):
-                    chunk_offset = chunk_info[0]
-                    byte_offset = chunk_info[2]
-                    size = chunk_info[3]
-
-                    if len(chunk_offset) != len(chunk_dims):
-                        pass
-                        msg = f"Unexptected chunk offset: {chunk_offset}"
-                        logging.error(msg)
-                        if not ctx["ignore_error"]:
-                            raise IOError(msg)
-                        return
-
-                    chunk_index = []
-
-                    for i in range(rank):
-                        chunk_index.append(chunk_offset[i] // chunk_dims[i])
-                    
-                    index = tuple(chunk_index)
-
-                    e = [byte_offset,size]
-
-                    if extend:
-                        e.append(s3path)
-                    e = tuple(e)
-
-                    chunkinfo_arr[index] = e
-
-                    return None
-
-                src.id.chunk_iter(update_chunktable_callback)
-
-            else: # Using old HDF5 version without H5Dchunk_iter
-                for i in range(num_chunks):
-                    chunk_info = src.id.get_chunk_info(i, spaceid)
-                    index = get_chunk_table_index(chunk_info.chunk_offset, chunk_dims)
-                    if not isinstance(index, tuple) or len(index) != rank:
-                        msg = f"Unexpected array_offset: {index} for dataset with rank: {rank}"
-                        logging.error(msg)
-                        if not ctx["ignore_error"]:
-                            raise IOError(msg)
-                        return
-                    e = [chunk_info.byte_offset, chunk_info.size]
-                    if extend:
-                        e.append(s3path)
-                    e = tuple(e)
-                    chunkinfo_arr[index] = e
-    else:
+    if not is_h5py(src):
```

Comment:
Is there a reason to remove the work done to the chunk table in the `is_h5py` case?

Reply:
yes, that might be a mistake.  Didn't test the scenario where this comes up - let me do that.

Reply:
Couldn't get anything to fail with the code removed.  I think this was initially for the case of hsload with extend and link but that's not supported